Composer/packages/client/src/recoilModel/dispatchers/project.ts

```diff
@@ -281,8 +304,23 @@ export const projectDispatcher = () => {
     try {
       await flushExistingTasks(callbackHelpers);
       set(botOpeningState, true);
-      await openRootBotAndSkillsByProjectId(callbackHelpers, projectId);
-
+      const { requiresMigrate } = await openRootBotAndSkillsByProjectId(callbackHelpers, projectId);
+      if (requiresMigrate) {
```

Comment:
Can we make this code and the code in line 236 be in a single location and reuse?

Reply:
added a forceMigrate function that encapsulates this for reuse!